src/client/mod.rs

```diff
@@ -17,7 +17,10 @@ use token::{Token, Lifetime, Refresh};
 
 /// OAuth 2.0 client.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct Client<P: Provider> {
+pub struct Client<P> {
```

Comment:
While the Provider specialization would show up in tooling to anyone trying to use Client::new, anyone casually interacting with a Client object wouldn't see the trait bound. I'm just imaging someone mousing over a Client in an IDE seeing a type signature of Client<P> and wondering what the P is since its not explicit in the struct signature.

What examples from std have unbounded structs with restricted impls?